internal/appconfig/validation.go

```diff
@@ -31,6 +31,11 @@ func (cfg *Config) Validate(ctx context.Context) (err error, extra_info string)
 		return err, extra_info
 	}
 
+	buildStrats := cfg.BuildStrategies()
+	if len(buildStrats) > 1 {
+		return fmt.Errorf("More than one build configuration found: [%s]", strings.Join(buildStrats, ", ")), extra_info
```

Comment:
I like it, much cleaner but failing hard here could be too much at first for users, we are going to break people's pipelines without notice.

We need some way of warning them and send an event to sentry to know how many are there : https://github.com/superfly/flyctl/blob/287d00975f600fa22ce6eeebda2f21ed8829f956/internal/appconfig/toplevelcheck.go#L33